lua/starfall/libs_sv/nextbot.lua

```diff
@@ -0,0 +1,598 @@
+	nb.IgniteCallbacks:remove(id)
+end
+
+--- Adds a callback function that will be run when the nextbot enters a new nav area.
+-- @server
+-- @param string callbackid The unique ID this callback will use.
+-- @param function callback The function to run when the NB enters a new nav area. The arguments are: (The NextBot, Old Nav Area, New Nav Area)
+function nb_methods:addNavChangeCallback(id, func)
+	local nb = nbunwrap(self)
+	checkpermission(instance, nb, "nextbot.addNavChangeCallback")
+	nb.NavChangeCallbacks:add(id, func)
+end
+
+--- Removes a nav area change callback function from the NextBot.
+-- @server
+-- @param string callbackid The unique ID of the callback to remove.
+function nb_methods:removeNavChangeCallback(id)
+	local nb = nbunwrap(self)
+	checkpermission(instance, nb, "nextbot.removeNavChangeCallback")
+	nb.NavChangeCallbacks:remove(id)
+end
+
+--- Sets a callback function that will be run when this nextbot touches another entity. Only 1 per NB. Setting a new callback will replace the old one.
+-- @server
+-- @param string callbackid The unique ID this callback will use.
+-- @param function callback The function to run when the NB touches another entity. The arguments are: (The NextBot, The entity the NB touched.)
+function nb_methods:addContactCallback(id, func)
+	local nb = nbunwrap(self)
+	checkpermission(instance, nb, "nextbot.addContactCallback")
+	nb.ContactCallbacks:add(id, func)
+end
+
+--- Removes the contact callback function from the NextBot if present.
+-- @server
+-- @param string callbackid The unique ID of the callback to remove.
+function nb_methods:removeContactCallback(id)
+	local nb = nbunwrap(self)
+	checkpermission(instance, nb, "nextbot.removeContactCallback")
+	nb.ContactCallbacks:remove(id)
+end
+
+--- Enable or disable ragdolling on death for the NextBot.
+-- @server
+-- @param boolean ragdollondeath Whether the nextbot should ragdoll on death.
+function nb_methods:ragdollOnDeath(bool)
+	local nb = nbunwrap(self)
+	checkpermission(instance, nb, "nextbot.ragdollOnDeath")
+	nb.RagdollOnDeath = bool
+end
+
+--- Sets the move speed of the NextBot.
+-- @server
+-- @param number newmovespeed NB's new move speed. Default is 200.
+function nb_methods:setMoveSpeed(val)	
+	local nb = nbunwrap(self)
+	checkpermission(instance, nb, "nextbot.setMoveSpeed")
+	nb.MoveSpeed = val
+end
+
+--- Gets the move speed of the NextBot.
+-- @server
+-- @return number NB's move speed.
+function nb_methods:getMoveSpeed(val)	
+	local nb = nbunwrap(self)
+	return nb.MoveSpeed
+end
+
+--- Sets the acceleration speed of the NextBot.
+-- @server
+-- @param number newaccel NB's new acceleration. Default is 400
+function nb_methods:setAcceleration(val)
+	local nb = nbunwrap(self)
+	checkpermission(instance, nb, "nextbot.setAcceleration")
+	nb.loco:SetAcceleration(val)
+end
+
+--- Gets the acceleration speed of the NextBot.
+-- @server
+-- @return number NB's acceleration value.
+function nb_methods:getAcceleration()
```

Comment:
Can you also add the one that returns a vector instead of number?

Reply:
It just returns a static acceleration value (which you can set with setAcceleration) that determines how fast the NPC will reach its MoveSpeed value. Idk what I'd return as a vector here.